src/components/vm/overview/vsock.jsx

```diff
@@ -0,0 +1,259 @@
+const SET_MANUALLY = _("Set manually");
+// There  are  several  reserved  addresses:
+// VMADDR_CID_ANY  (-1U)  means  any  address  for  binding;
+// VMADDR_CID_HYPERVISOR (0) is reserved for services built into the hypervisor;
+// VMADDR_CID_LOCAL (1) is the well-known address for local communication (loopback);
+// VMADDR_CID_HOST (2) is the well-known address of the host.
+const MIN_VSOCK_CID = 3;
+
+function getVsockUsageMessage(vmName, connectionName, vms, auto, address) {
+    if (auto)
+        return;
+
+    const vmsUsingCIDAddress = vms
+            .filter(vm => !(vmName === vm.name && connectionName === vm.connectionName) && vm.vsock.cid.auto === "no" && vm.vsock.cid.address === String(address))
+            .map(vm => vm.name);
+    if (vmsUsingCIDAddress.length === 0)
+        return;
+
+    const vmsUsingCIDAddressString = vmsUsingCIDAddress.join(", ");
+
+    return cockpit.format(_("This context identifier is already used by $0. Assigning the same CID is possible, but the VMs won't be able to run simultaneously."), vmsUsingCIDAddressString);
+}
+
+function getNextAvailableVsockCID(vmName, connectionName, vms) {
+    let availableAddress = MIN_VSOCK_CID;
+
+    const vmsCIDAddresses = vms.filter(vm => !(vmName === vm.name && connectionName === vm.connectionName) && vm.vsock.cid.auto === "no")
+            .map(vm => Number(vm.vsock.cid.address))
+            .sort();
+
+    for (let i = 0; i < vmsCIDAddresses.length; i++) {
+        const addressInUse = vmsCIDAddresses[i];
+        if (availableAddress === addressInUse) {
+            availableAddress++;
+            i = 0;
+        }
+    }
+
+    return availableAddress;
+}
+
+export const VsockModal = ({ vm, vms, vmVsockNormalized, isVsockAttached, idPrefix }) => {
+    const [dialogError, setDialogError] = useState();
+    const [auto, setAuto] = useState(vm.vsock.cid.auto ? vmVsockNormalized.auto : true);
+    const [address, _setAddress] = useState(vmVsockNormalized.address || getNextAvailableVsockCID(vm.name, vm.connectionName, vms));
+    const [inProgress, setInProgress] = useState(false);
+
+    const setAddress = (value) => {
+        // Allow empty string
+        if (value === "") {
+            _setAddress(value);
+            return;
+        }
+
+        _setAddress(parseInt(value));
+    };
+
+    const onBlur = (value) => {
+        if (value < MIN_VSOCK_CID)
+            value = MIN_VSOCK_CID;
+
+        _setAddress(value);
+    };
+
+    const Dialogs = useDialogs();
+
+    const save = () => {
+        setInProgress(true);
+        return domainSetVsock({
+            connectionName: vm.connectionName,
+            vmName: vm.name,
+            hotplug: vm.state === "running",
+            permanent: vm.persistent,
+            auto: auto ? "yes" : "no",
+            address,
+            isVsockAttached,
+        })
+                .then(Dialogs.close, exc => setDialogError({ text: _("Failed to configure vsock"), detail: exc.message }))
```

Comment:
Please use "Vsock" consistently.

Reply:
Fixed, now it uses "vsock" everywhere

Reply:
Commented my opinion here: https://github.com/cockpit-project/cockpit-machines/pull/969#discussion_r1164467065